### ISSUE TITLE
DeserializingConsumer does not return a regular Message

### ISSUE THREAD

**Issue by InterferencePattern** [confluentinc/confluent-kafka-python#1800]
Description
===========
The `Message` object returned by `DeserializingConsumer.value()` doesn't appear to match the [signature](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Message.value)- it can return a `dict`, for example, and not only a `str` or `bytes`. 

How to reproduce
================
1. Use `AvroDeserializer` to create `DeserializingConsumer`
2. Use `.poll()` method to return message
3. `Message.value()` should contain deserialized dictionary instead of `str` or `bytes`

Checklist
=========
Please provide the following information:

 - [x] confluent-kafka-python and librdkafka version (`confluent_kafka.version()` and `confluent_kafka.libversion()`): ('2.3.0', 33751040)
 - [ ] Apache Kafka broker version:
 - [ ] Client configuration: `{...}`
 - [ ] Operating system:
 - [ ] Provide client logs (with `'debug': '..'` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by InterferencePattern** [comment]
This issue isn't that it doesn't work- it does. But the `Message.value()` return from a DeserializingConsumer is a dictionary, right? If so, this doesn't conform to the output in the `Message` documentation and should be its own class.

**Comment by pranavrth** [comment]
> But the Message.value() return from a DeserializingConsumer is a dictionary, right?

No, it depends on the value serializer that is provided. If no serializer is provided, it is same as what a normal consumer would return i.e. byte/str.

**Comment by InterferencePattern** [comment]
Okay, I've narrowed the Issue title to indicate that this is true for the `AvroDeserializer`.

**Comment by pranavrth** [comment]
I would let @rayokota handle from here whose team deals with Schema Registry.
